config/config_loader.go

```diff
@@ -86,12 +100,48 @@ func init() {
 		log.Printf("[providerInit] %#v", errPro)
 		providerConfig = nil
 	} else {
+		// Check if there are some important key fields missing,
+		// if so, we set a default value for it
+		setDefaultValue(providerConfig)
 		// Even though baseConfig has been initialized, we override it
 		// because we think read from config file is correct config
 		baseConfig = &providerConfig.BaseConfig
 	}
 }
 
+// setDefaultValue set default value for providerConfig or consumerConfig if it is null
+func setDefaultValue(target interface{}) {
```

Comment:
Can you use annotation in [this line](https://github.com/apache/dubbo-go/blob/274718b650e71ee55a832203e87088f9762d4ff7/config/consumer_config.go#L53) for default value?

Reply:
> Can you use annotation in [this line](https://github.com/apache/dubbo-go/blob/274718b650e71ee55a832203e87088f9762d4ff7/config/consumer_config.go#L53) for default value?

in go we can use annotation as Java?

Reply:
let me see see and study study. hhhh